metrics-operator/controllers/analysis/controller.go

```diff
@@ -136,9 +144,69 @@ func (a *AnalysisReconciler) evaluateObjectives(ctx context.Context, res map[str
 		analysis.Status.Warning = true
 	}
 	analysis.Status.Pass = eval.Pass
+	go a.reportResultsAsPromMetric(eval, analysis)
 	return a.updateStatus(ctx, analysis)
 }
 
+func (a *AnalysisReconciler) reportResultsAsPromMetric(eval evalType.AnalysisResult, analysis *metricsapi.Analysis) {
+	f := analysis.Spec.From.String()
+	t := analysis.Spec.To.String()
+	labelsAnalysis := prometheus.Labels{
+		"name":      analysis.Name,
+		"namespace": analysis.Namespace,
+		"from":      f,
+		"to":        t,
+	}
+	if m, err := a.Metrics.AnalysisResult.GetMetricWith(labelsAnalysis); err == nil {
+		m.Set(eval.GetAchievedPercentage())
+	} else {
+		a.Log.Error(err, "unable to set value for analysis result metric")
+	}
+	// expose also the individual objectives
+	for _, o := range eval.ObjectiveResults {
+		name := o.Objective.AnalysisValueTemplateRef.Name
+		ns := o.Objective.AnalysisValueTemplateRef.Namespace
+		labelsObjective := prometheus.Labels{
+			"name":               name,
+			"namespace":          ns,
+			"analysis_name":      analysis.Name,
+			"analysis_namespace": analysis.Namespace,
+			"key_objective":      fmt.Sprintf("%v", o.Objective.KeyObjective),
+			"weight":             fmt.Sprintf("%v", o.Objective.Weight),
+			"from":               f,
+			"to":                 t,
+		}
+		if m, err := a.Metrics.ObjectiveResult.GetMetricWith(labelsObjective); err == nil {
+			m.Set(o.Value)
+		} else {
+			a.Log.Error(err, "unable to set value for objective result metric")
+		}
+	}
+}
+
+//nolint:ineffassign,staticcheck
+func SetupMetric() (m Metrics, err error) {
```

Comment:
I would move this function is some other place, this sets up metrics, it has nothing to do with controller functionality

Reply:
It kinda of does. 
I don't like this from Prometheus: you need to create the vectorized gauges with a set of labels and when recording a value for the metric, you must specify a map with exactly the same set of keys of the labels. Therefore there is a coupling between init of the metrics and the controller exporting values for said metrics.
I thought about having everything hide behind the standard `NewAnalysisController` so we could have more encapsulation, but I opted for the current solution to keep all controller consistent wrt their initialization. I am open for suggestions/opinions here :)

Reply:
makes sense, we can keep it here and move it when we have a nicer solution